components/webui/imports/ui/SearchView/datetime.js

```diff
@@ -80,45 +65,149 @@ const computeTimeRange = (token) => () => {
             TIME_RANGE_MODIFIER.TO_DATE,
         ].includes(modifier);
 
-        endTime = (true === isEndingNow) ?
-            DateTime.utc() :
-            DateTime.utc().minus({[unit]: amount}).endOf(unit);
-        beginTime = (true === isBeginStartOfUnit) ?
-            endTime.startOf(unit) :
-            endTime.minus({[unit]: amount});
+        end = (true === isEndingNow) ?
+            dayjs.utc() :
+            dayjs.utc().subtract(amount, unit)
+                .endOf(unit);
+        begin = (true === isBeginStartOfUnit) ?
+            end.startOf(unit) :
+            end.subtract(amount, unit);
     }
 
     return {
-        begin: dateTimeToDateWithoutChangingTimestamp(beginTime),
-        end: dateTimeToDateWithoutChangingTimestamp(endTime),
+        begin,
+        end,
     };
 };
 
+const DEFAULT_TIME_RANGE = computeTimeRange(
+    `${TIME_UNIT.ALL}_${TIME_RANGE_MODIFIER.NONE}_0`,
+);
+
+// TODO: Switch date pickers so we don't have to do this hack
 /**
- * Changes the timezone of a given Date object to UTC without changing the time.
+ * Converts a Dayjs object in UTC, to a JavaScript Date object that represents the same date and
+ * time, but aligned with the local timezone.
+ * This is achieved without altering the original timestamp values (year, month, day, hour,
+ * minute, second, and millisecond).
  *
- * @param {Date} date Date object to convert to UTC
- * @returns {Date} A new Date object with the same time values in UTC timezone
+ * @param {dayjs.Dayjs} utcDatetime
+ * @return {Date | undefined} The corresponding Date object, or `undefined` when `utcDatetime`
+ * is `undefined`.
  */
-const changeTimezoneToUtcWithoutChangingTime = (date) => {
-    return new Date(Date.UTC(
-        date.getFullYear(),
-        date.getMonth(),
-        date.getDate(),
-        date.getHours(),
-        date.getMinutes(),
-        date.getSeconds(),
-        date.getMilliseconds(),
-    ));
+const convertUtcToSameLocalDate = (utcDatetime) => {
+    const localTz = dayjs.tz.guess();
+    return utcDatetime?.tz(localTz, true).toDate();
 };
 
-const DEFAULT_TIME_RANGE = computeTimeRange(
-    `${TIME_RANGE_UNIT.ALL}_${TIME_RANGE_MODIFIER.NONE}_0`,
-);
+/**
+ * Converts a JavaScript Date object in local timezone, to a Dayjs object that represents the same
+ * date and time, but aligned with the UTC.
+ *
+ * @param {Date} localDate
+ * @return {dayjs.Dayjs} The corresponding Dayjs object
+ */
+const convertLocalToSameUtcDatetime = (localDate) => {
+    return dayjs(localDate)?.utc(true);
```

Comment:
`?` removed.

As discussed above, the original thoughts were to use those as wrappers. Since we have control over the usages, I agree the inputs & outputs should be handled explicitly.